### ISSUE TITLE
Make manual test results available in Cross Platform Release Notes

### ISSUE THREAD

**Issue by rfennell** [rfennell/AzurePipelines#959]
At present only test results directly published by a task in a build e.g. VSTest are available to the release notes template.

Make it possible to also access any manual test results associated with a build via an Azure DevOps Test Plan

See #958 for discussion of the background to this request

**Comment by rfennell** [comment]
It should be noted that this feature may generate some strange results in 'real world' usage. This is because manual Test Plans tend to be run after a build has completed i.e. after the release notes are created.

Any practical us of this feature does assume that the pipeline is making use of multi-stage YAML, so that we have a flow similar to

1. A stage where the build can be run to produced the system under test,
2. A stage to deploy the system so it can be tested
3. A stage to run the Azure DevOps Manual Test suite
4. A final stage to produce the release notes.

**Comment by rfennell** [comment]
I had a few minutes today and have done some initial research.

- Given a build ID, I need to make the query to `_apis/test/runs?api-version=6.0&buildids=<id>`. This will return all the test plan runs associated with the build ID.
- Each test plan run returned includes a URL to the API in the form `/_apis/test/Runs/1000052`. The return from this call is the core of what is required in release notes i.e

>"postProcessState": "Complete",
"totalTests": 3,
"incompleteTests": 0,
"notApplicableTests": 0,
"passedTests": 3,
"unanalyzedTests": 0,
- The results of each test within this test run can be found using the call `_apis/test/Runs/1000052/results` which returns an array of test results.

Note: these are raw API calls, in reality, the Node API will be used to access these endpoints.

It is proposed a new root object is added to those available to the template 

- ManualTestPlans [an array of objects returned from `/_apis/test/Runs/1000052` style calls]
    - TestResults [an array of objects returned from `/_apis/test/Runs/1000052/results` style calls]

**Comment by rfennell** [comment]
A progress report, I have a working first version, this is being regression tested and will be released as 3.46.x

Getting the details of the manual tests was not too hard. The problem turned out to be getting the details of the test configurations the individual tests in the test run were run against. This detail is needed to differentiate tests if you run the same test against multiple configurations. Without this detail, you see the same test multiple times in the test run with no context. The major issue is the required [Configuration REST](https://docs.microsoft.com/en-us/rest/api/azure/devops/test/configurations/get?view=azure-devops-rest-5.0) is only in preview, so is not available in the Azure DevOps Node SDK. Hence, a native REST call must be made. 

This all means I have had to add two new template objects

- manualtests - the array of manual Test Plan runs associated with any of the builds linked to the release 
- manualTestConfigurations - the array of manual test configurations

Also, the template build array, used when the task is called in the release, has been extended with a manualtests property for each build.  

These new objects can be used in templates to produce a simple list of test runs

```
## Manual Test Plans
| Run ID | Name | State | Total Tests | Passed Tests |
| --- | --- | --- | --- | --- |
{{#forEach manualTests}}
| [{{this.id}}]({{this.webAccessUrl}}) | {{this.name}} | {{this.state}} | {{this.totalTests}} | {{this.passedTests}} |
{{/forEach}}
```

If you wish to list all the individual tests you need to lookup the configuration for each test to get the configuration that was used.

```
## Manual Test Plans with test details
{{#forEach manualTests}}
### [{{this.id}}]({{this.webAccessUrl}}) {{this.name}} - {{this.state}}

| Test | Outcome | Configuration |
| - | - | - |
{{#forEach this.TestResults}}
| {{this.testCaseTitle}} | {{this.outcome}} | {{#with (lookup_a_test_configuration ../../manualTestConfigurations this.configuration.id)}} {{this.name}} {{/with}} |
{{/forEach}}
{{/forEach}}
``` 

These samples only show a few of the properties available on these new object. To see all the options use the `{{ json manualTests }}`
